controllers/storagecluster/storageconsumer.go

```diff
@@ -136,15 +136,34 @@ func (s *storageConsumer) ensureCreated(r *StorageClusterReconciler, storageClus
 			availableServices,
 		)
 		resourceMap := util.WrapStorageConsumerResourceMap(data)
-		if backwardCompatibleInfo.Pre4_19InternalConsumer == "" {
+		switch backwardCompatibleInfo.Pre4_19DeploymentMode {
+		case "":
```

Comment:
Don't handle both compatibility mode and non compatibility mode using the same switch. One decision need to be made before the other on a separate context.
```
if !compatabilityMode {
} else {
     switch compatability mode type
}
```